components-registry-service-server/src/main/kotlin/org/octopusden/octopus/components/registry/server/service/impl/ComponentRegistryResolverImpl.kt

```diff
@@ -218,14 +218,17 @@ class ComponentRegistryResolverImpl(
      */
     override fun getComponentsCountByBuildSystem(): EnumMap<BuildSystem, Int> {
         val components = getComponents()
-        val result = EnumMap<BuildSystem, Int>(BuildSystem::class.java)
+        val result = EnumMap<BuildSystem, Set<String>>(BuildSystem::class.java)
         components.forEach { component ->
             component.moduleConfigurations.forEach { moduleConfig ->
+                if (moduleConfig.componentDisplayName?.endsWith("(archived)") == true) {
```

Comment:
Do we really need == true?

Reply:
moduleConfig.componentDisplayName — is String? (nullable string).